internal/eth/txn.go

```diff
@@ -63,6 +63,21 @@ type TxnReceipt struct {
 	TransactionIndex  *ethbinding.HexUint   `json:"transactionIndex"`
 }
 
+// TxnInfo is the detailed transaction info returned by eth_getTransactionByXXXXX
+type TxnInfo struct {
+	BlockHash        *ethbinding.Hash      `json:"blockHash,omitempty"`
+	BlockNumber      *ethbinding.HexBigInt `json:"blockNumber,omitempty"`
+	From             *ethbinding.Address   `json:"from,omitempty"`
+	To               *ethbinding.Address   `json:"to,omitempty"`
+	Gas              *ethbinding.HexBigInt `json:"gas"`
+	GasPrice         *ethbinding.HexBigInt `json:"gasPrice"`
+	Hash             *ethbinding.Hash      `json:"hash"`
+	Nonce            *ethbinding.HexUint   `json:"nonce"`
+	TransactionIndex *ethbinding.HexUint   `json:"transactionIndex"`
+	Value            *ethbinding.HexBigInt `json:"value"`
+	Input            *ethbinding.HexBytes  `json:"input"`
```

Comment:
How do I know the types for these values returned by geth? Particularly on some of the BigInt vs Uint I just guessed... should they all just be BigInt?

Reply:
The official reference is here: https://eth.wiki/json-rpc/API#eth_gettransactionbyhash
However, I find going straight to the code best:
https://github.com/ethereum/go-ethereum/blob/9ada4a2e2c415e6b0b51c50e901336872e028872/internal/ethapi/api.go#L1264-L1284

One quick conversation to have here before we merge, is that ethconnect does try and make various returns more friendly in other places. Giving values in non-hex decimal strings, as well as hex.
I'm wondering if we should be consistent here with the mapping.

See: https://github.com/hyperledger-labs/firefly-ethconnect/blob/984cba49f715ea3f5535261d4b38564816065a0a/internal/messages/messages.go#L141-L163

Reply:
Yea the docs just listed everything as `QUANTITY` which was why I couldn't tell the expected sizes. Thanks for the code pointer! Looks like I got pretty close, except that `Gas`, `Nonce`, and `TransactionIndex` can all be `Uint64`. I'll update.

I guess it makes sense to be consistent with how the fields are formatted between this and the `TransactionReceipt` you linked as well.